ios/Classes/AblyStreamsChannel.m

```diff
@@ -0,0 +1,138 @@
+    return;
+  }
+  
+  _streams = [NSMutableDictionary new];
+  _listenerArguments = [NSMutableDictionary new];
+  FlutterBinaryMessageHandler messageHandler = ^(NSData* message, FlutterBinaryReply callback) {
+    FlutterMethodCall* call = [self->_codec decodeMethodCall:message];
+    NSArray *methodParts = [call.method componentsSeparatedByString:@"#"];
+    
+    if (methodParts.count != 2) {
+      callback(nil);
+      return;
+    }
+    
+    NSInteger keyValue = [methodParts.lastObject integerValue];
+    if(keyValue == 0) {
+      callback([self->_codec encodeErrorEnvelope:[FlutterError errorWithCode:@"error" message:[NSString stringWithFormat:@"Invalid method name: %@", call.method] details:nil]]);
+      return;
+    }
+    
+    NSNumber *key = [NSNumber numberWithInteger:keyValue];
+    
+    if ([methodParts.firstObject isEqualToString:@"listen"]) {
+        [self listenForCall:call withKey:key usingCallback:callback andFactory:factory];
+    } else if ([methodParts.firstObject isEqualToString:@"cancel"]) {
+        [self cancelForCall:call withKey:key usingCallback:callback andFactory:factory];
+    } else {
+      callback(nil);
+    }
+  };
+  
+  [_messenger setMessageHandlerOnChannel:_name binaryMessageHandler:messageHandler];
+}
+
+- (void) reset{
+    for (NSString* key in self->_streams) {
+        AblyStreamsChannelStream *stream = self->_streams[key];
+        [stream.handler onCancelWithArguments:nil];
+        [_streams removeObjectForKey:key];
+        [_listenerArguments removeObjectForKey:key];
+    }
+}
+
+- (void)listenForCall:(FlutterMethodCall*)call withKey:(NSNumber*)key usingCallback:(FlutterBinaryReply)callback andFactory:(NSObject<FlutterStreamHandler> *(^)(id))factory {
+  AblyStreamsChannelStream *stream = [AblyStreamsChannelStream new];
+  stream.sink = ^(id event) {
+    NSString *name = [NSString stringWithFormat:@"%@#%@", self->_name, key];
+    
+    if (event == FlutterEndOfEventStream) {
+      [self->_messenger sendOnChannel:name message:nil];
+    } else if ([event isKindOfClass:[FlutterError class]]) {
+      [self->_messenger sendOnChannel:name
+                              message:[self->_codec encodeErrorEnvelope:(FlutterError*)event]];
+    } else {
+      [self->_messenger sendOnChannel:name message:[self->_codec encodeSuccessEnvelope:event]];
+    }
+  };
+  stream.handler = factory(call.arguments);
+  
+  [_streams setObject:stream forKey:key];
+  [_listenerArguments setObject:call.arguments forKey:key];
+  
+  FlutterError* error = [stream.handler onListenWithArguments:call.arguments eventSink:stream.sink];
+  if (error) {
+    callback([self->_codec encodeErrorEnvelope:error]);
+  } else {
+    callback([self->_codec encodeSuccessEnvelope:nil]);
+  }
+}
+  
+- (void)cancelForCall:(FlutterMethodCall*)call withKey:(NSNumber*)key usingCallback:(FlutterBinaryReply)callback andFactory:(NSObject<FlutterStreamHandler> *(^)(id))factory {
```

Comment:
The various `self->` prefixes for ivar access are superfluous in this method implementation and just add noise. You only need explicit `self` when dealing with blocks.

Reply:
Addressed in 86deac257bdc6054137ddedbfe31e85cd716412c